application/app.py

```diff
@@ -93,28 +95,62 @@ def get_messages():
 
 @app.route("/api/message", methods=['POST'])
 def incoming_message():
+    """ Respond to an incoming SMS message.
+
+        First, checks to see if the message can be recognized as one of the
+        commands in the SMS API and responds appropriately if so. If not, this
+        will try to determine whether the message is in response to another
+        message or in a current event, in order to determine how to thread the
+        message.  """
     from_number = request.values.get('From', None)
     body = request.values.get('Body', None)
     timestamp = int(time.time())
     user = User.from_number(from_number)
     event = 0
+    # The ID of the last message sent to the user is stored on the user object.
+    # If that exists and refers to an active event, we assume this message
+    # related to the samne event.
```

Comment:
Totally agree that this logic leaves a lot to be desired, but I couldn't really think through what elements we'd need to preserve about the user's state to do this better. I'm hoping we'll come up with an solution for that as we build out the text interface a little more. Maybe we can add a json blob field to the user model that stores some basic info of the recent messages they've recieved and sent, so that we can do this more intelligently than guessing what message they might be responding to.

Fixed the typo in 7b584ae.